src/core/expr/fexpr_extend_remove.h

```diff
@@ -0,0 +1,42 @@
+//------------------------------------------------------------------------------
+// Copyright 2022 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#ifndef dt_EXPR_FEXPR_EXTEND_REMOVE_h
+#define dt_EXPR_FEXPR_EXTEND_REMOVE_h
+#include "expr/fexpr_func.h"
+namespace dt {
+namespace expr {
+
+class FExpr_Extend_Remove : public FExpr_Func {
+  private:
+    ptrExpr arg_;
+    ptrExpr values_;
+    bool extend_;
```

Comment:
@oleksiyskononenko could you kindly have a look at my latest commit. where am I getting it wrong with my knowledge of templates

Reply:
Sure, the way this template should be implemented is exactly the same as for many other classes we have recently created. Could you please explain me what is the issue you are facing?

Reply:
this is the error I am getting: 

```py
src/core/expr/fexpr_extend_remove.cc:28:1: error: invalid use of template-name ‘dt::expr::FExpr_Extend_Remove’ without an argument list
   28 | FExpr_Extend_Remove::FExpr_Extend_Remove(ptrExpr&& arg, ptrExpr&& other) :
      | ^~~~~~~~~~~~~~~~~~~
```

my google fu hasnt fetched me anything helpful

Reply:
i basically add the `template<bool EXTEND>` to `fexpr_extend_remove.h` and `fexpr_extend_remove.cc`

Reply:
Since you keep template declarations in the `.h` file and all the definitions in `.cc`, you need to explicitly specify which `EXTEND` values, i.e. `true` and `false`, you're implementing outside of the header file. Also, definitions of the class methods should all include the template parameter. Take a look at the changes I've just pushed.

Reply:
Thank you @oleksiyskononenko ... It was hard for me to decipher this solution based on the error message. Thanks agaon

Reply:
Yeah, error messages for templates are not really helpful. The standard way to implement it is to put everything in the header file. Once you split the declarations/definitions, some issues may arise. If you need more info on this, google for “C++ template instantiation”.